### ISSUE TITLE
"if no switch condition is met for" locks me to change scene forever

### ISSUE THREAD

**Issue by Barabba11** [WarmUpTill/SceneSwitcher#306]
Hi! I appreciate your plugin, I would like to "st-by" OBS, stopping decoding video streams input, if it not in use for some minutes, for this I'm using a scene with an image only where written "choose your scene", the scene is selected by websockets. Actually at time out I'm not abe to chose another scene, it returns automatically at the image scene, I would like it would be free to change scenes and the timer restarts automatically every time a scene is changed, being resetted. How is possible to do? I've tried to use macros, but seems not possible.
Thank you a lot!

**Comment by WarmUpTill** [comment]
Hi,
I am sorry, but I do not understand what you are trying to achieve exactly.
Can you describe the condition, which should trigger a scene change, in a bit more detail?
So what do you mean by "if it not in use for some minutes"? 
(e.g. the video output has not changed / there was no scene change / there were no user inputs)

Thanks!

And just to clarify:
The "if no condition is met for ... switch to ... " does exactly what the name suggestions.
If no scene switch condition is true for a certain period of time switch to the specified scene.
It does not look at any other factors.

**Comment by Barabba11** [comment]
thanks for quick reply, with pleasure I try to explain better.
Actually it locks, whrn the timeout comes I can't change the scene, better I say it changes for a fraction of second, then it goes back to the scene I selected. Maybe a bug? 
The goal is to switch to an image if nobody usng obs, to avoid cpu usage on decoding. When an input comes (I'm using websockets, but I can also decide to manually click on other scenes with mouse in the obs main interface,scene list) I should be free to switch it and the timer is resetting, displaying again the picture if no inputs. Each time I swtich a scene the timer should reload, so the idle time will be always the same. For example
the timeout is 1 hour, 
1) I select scene and do nothing> one hour> image > I change scene and do nothing > one hour >image
2) I select scene and do nothing> 30 minutes and I cange scene > one hour> image

**Comment by WarmUpTill** [comment]
Ah now I understand - so you basically want something like "if scene was not changed for X -> switch to scene "image".
I added such functionality with the linked PR - so you can configure something similar to this on the macro tab:

![example](https://user-images.githubusercontent.com/19472752/133880039-3aff7a72-9a92-46ff-b1c4-2d189ba88647.PNG)

You can find a build here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/1248070015

Let me know if that works as expected! :)
(Remember to disable the "if no condition is met for ..." setting)

**Comment by Barabba11** [comment]
thank you mate, sorry to say you it doesn't work. The problem I suppose is that it doesn't "catch" exactly 20.00 seconds, so it doesn't start. I've then selected "at most" but the same problem, the scene is contiuniusly forced to the image. The problem here is that the timer is not reset after the scene changed, I guess.
Can you introduce into the macro a command to reset the timer? When a scene is changed, timer is reset
Thank you

**Comment by WarmUpTill** [comment]
Glad to hear that it is working!

>it was necessary to restart plugin after macro defined, esle it was not working properly.

Hm strange.
What was the behavior you you observed, when it was not working properly?
Are you able to reproduce the issue?
If so how?

**Comment by Barabba11** [comment]
It was doing as before, can't change the scene, it returns to the imagine. But don't switch was selected. I've tried to reproduce changing different paramenters, no way. All work as expected.
After match don't switch scenes for was placed to 1, I put it on 0, maybe was that.. but I don't think so. Then I stopped and and started of the plugin.
Let's forget about this for now, if there is any issue incoming I'll try to report you as best I can. Thank you!
